[caffe2] Remove cxx override to c++17

thevinster opened the pull request:
Summary: Allow the user to specify the cxx version to use when compiling. For applications that compile with C++20, we wish to also compile this library with C++20 to avoid subtle ODR violations with using different library standards.

Test Plan: Built the project successfully.

Reviewed By: smeenai

Differential Revision: D48636406

davidberard98 reviewed (approved):
wait for tests to pass, otherwise LGTM

thevinster committed:
[caffe2] Remove cxx override to c++17 (#108687)

Summary:

Allow the user to specify the cxx version to use when compiling. For applications that compile with C++20, we wish to also compile this library with C++20 to avoid subtle ODR violations with using different library standards.

Test Plan: Built the project successfully.

Reviewed By: smeenai, davidberard98

Differential Revision: D48636406
